app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -161,6 +162,14 @@
     private boolean cancelled;
     private ContainerFragment<EntityListAdapter> containerFragment;
 
+    // Function handler for handling XPath evaluation of the function here().
+    // Although only one instance is created, which is used by NodeEntityFactory,
+    // every instance of EntitySelectActivity registers itself (one at a time)
+    // to listen to the handler and refresh whenever a new location is obtained.
+    public static final HereFunctionHandler hereFunctionHandler = new HereFunctionHandler();

```

Comment:
I would make this private and expose it via a getter method. Allows you to better control the internal implementation later on, in case you want to return new instances each time or do other crazy stuff.